src/github.com/getlantern/flashlight/genconfig/checkfallbacks.go

```diff
@@ -0,0 +1,166 @@
+// Utility for checking if fallback servers are working properly.
+// It outputs failing servers info in STDOUT.  This allows this program to be
+// used for automated testing of the fallback servers as a cron job.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/getlantern/golog"
+
+	"github.com/getlantern/flashlight/client"
+)
+
+var (
+	help          = flag.Bool("help", false, "Get usage help")
+	fallbacksFile = flag.String("fallbacks", "fallbacks.json", "File containing json array of fallback information")
+	verbose       = flag.Bool("verbose", false, "Verbose output for debugging")
+	numConns      = flag.Int("connections", 1, "Number of simultaneous connections")
+
+	expectedBody = "Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"
+)
+
+var (
+	log = golog.LoggerFor("checkfallbacks")
+)
+
+type fallbackServer map[string]interface{}
+
+func main() {
+	flag.Parse()
+
+	if *help {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	numcores := runtime.NumCPU()
+
+	if *verbose {
+		log.Debugf("Using all %d cores on machine", numcores)
+	}
+	runtime.GOMAXPROCS(numcores)
+
+	fallbacks := make(chan fallbackServer)
+	loadFallbacks(fallbacks)
+	testFallbacks(fallbacks)
+}
+
+// Load the fallback servers list file. Failure to do so will result in
+// exiting the program.
```

Comment:
Not a big deal, but maybe it would be best for this to take a `filename string` and return `([]string, err)`, so it would be simpler to unit test.  Then let the caller handle flags and either exit on error or feed the unmarshalledFallbacks to the channel.  I'd push those basically-untestable tasks to the `main` function.

This is probably only worth changing when we actually get around to writing unit tests for this.